src/ripple/app/main/Application.cpp

```diff
@@ -1599,7 +1599,7 @@ ApplicationImp::loadLedgerFromFile (
         {
             Json::Value& entry = ledger.get()[index];
 
-            if (!entry.isObject())
+            if (!entry.isObjectOrNull())
```

Comment:
I struggled with this one too.  It looks to me like `SetHex` of "" is going to return true and thus not error out.  But if a null `entry` is _supposed_ to error out, then I'd also love to catch it at line 1602.

Reply:
Well, we're reading a ledger out of a file here.  To do so we're walking the consecutive entries in an array.

1. One of those entries came up empty, which is how we got the null.
2. Then, because we got an empty array entry, we'll infer that the `index` (which is not explicit) is zero.
3. On line 1622 we see that `uIndex.isZero() == true` so we log "Invalid entry in ledger" and return `nullptr`.

I think we're better off immediately rejecting a null array entry.